MiniStation 2020

Syuff opened the pull request:
## About The Pull Request

Updated MiniStation to current tgstation code.

Replaced cloning with organ harvester and limb grower.

Replaced xenobiology lab with robotics lab.

Minor map changes and major bug fixes.

## Why It's Good For The Game

MiniStation fills a hole in the current map rotation for low population servers and private servers/LAN parties. The changes made to MiniStation also serve as an experiment for stations lacking cloning.

This map needs additional testing to find any additional bugs and balance issues. I have not yet tested the interactions with game modes other than extended and I hope that this pull request allows some servers to test it.

I welcome any and all contributions to this map and hope that some of you find this useful.

Syuff committed:
update _basemap to include ministation

Syuff committed:
Add ministation.json

Syuff committed:
Add MiniStation map and job changes

Syuff committed:
Update map_changes to include ministation job_changes

HomocidlePancakes commented:
How about we don’t get ahead of ourselfs and wait for the cloning thingamajig to get resolved?

Syuff commented:
> How about we don’t get ahead of ourselfs and wait for the cloning thingamajig to get resolved?

I believe cloning would be overkill for a small station such as this. I am not committed to the idea of removing it but for this sized map I thought it appropriate.

HomocidlePancakes commented:
Why would be an overkill? Manpower is limited - there may be noone to do surgery.

Syuff commented:
> Why would be an overkill? Manpower is limited - there may be noone to do surgery.

The small map with limited private areas makes murdering and disposing of bodies harder for antags. I thought that for a population of less than 10 people if cloning exists it would be too hard for antags. I will see what other people's opinions are with play testing and may revert the change.

HomocidlePancakes commented:
1)10 pop map meant for LANs and private servers and you worry antags.

2)I’ve just looked at the mapdiffbot and not gonna lie - MiniStation is literally handicaped. It lacks job equipment, it’s barren and overall ugly to look at. You’d be better off with redoing Omega. Just add viro and genetics there.

Syuff commented:
Ministation is basic and barren, yes, it is why I chose it over Omegastation, I didn't want to create Pubby 2.0. The other day I was thinking to myself how I would like to introduce this game to some friends, but currently the only server in Australia crashes my byond client whilst vanilla tg runs fine. I looked into what small maps existed previously and might be easy/fun for a new player. I remember spending almost entire rounds walking through maintenance tunnels trying to get to where I wanted and not having any fun.

I believe this map caters for low population for those who don't have the option of playing high pop servers. The basic and smaller map means more interactions with other crew members and less isolation than the other maps.

I decided to fix this map because I believe it fills the niche. I realise this is probably not an exciting addition for veteran players and project contributors, I just want to share some of the work I did because I think there may be some people who would appreciate this style of map.

There are still some issues that might need fixing but I don't know enough about it. The vents are not spawned in the modern style but they still work, some of the pipe icons in the map editor look weird and I don't know if that is okay or not.

HomocidlePancakes commented:
>I believe this map caters for low population for those who don't have the option of playing high pop servers. The basic and smaller map means more interactions with other crew members and less isolation than the other maps.

Again, Mini is handicapped. It has half the activities missing and it’s design is just outdated. You can claim that it encourages “player interaction” as much as you want, but it won’t fix Minis issues.

>I decided to fix this map because I believe it fills the niche. I realise this is probably not an exciting addition for veteran players and project contributors, I just want to share some of the work I did because I think there may be some people who would appreciate this style of map.

Nostalgia is nice, but it doesn’t justify everything.

ArcaneMusic commented:
I think the reasoning is sound, ministation has and always will be a bit barren, but barring it for a smaller population works fine IMO.That being said, there's plenty of room for improvement if you care for some constructive criticism:

- The floorspace around ministation is exceptionally barren. You could always liven up some of the hallways with some floor decals, or some different floor tile types to give some of the more plain areas a distinct feel.
- You're missing an experimentor and a BEPIS somewhere in science 👀 
- The Dorms-Janitor-Recycler space on Mini has always been a bit redundant. Since there's no disposals system, I'd say give it the kilo treatment and roll the recycler into the Janitor's closet, and then use that new left-over space into expanding dorms a bit.
- I have literally no idea what's going on with that ore-smelting-stacking thing in cargo. Since we're all about the ORM now, might be best to nix that and turn it into a warehouse area?
- 2 things with the armory, first is replace the armory's walls with Rwalls since that's an instant access from space. Second is that you'll likely need more officer lockers. Easiest option there is just build into the mait area right next to it and throw in some extra lockers. (Also don't forget that the new traitor objective item is the trenchcoat, not the reflector vest!)
- Now this is a personal gripe more than anything else: The asteroids to the north west of the station were originally meant as a stand-in for the asteroid so that you could get minerals on z-level without as much effort. The downside is that in modern maps we have the lavaland shuttle. You COULD potentially nix the lavaland shuttle for this map and really expand upon the meteor here to load you up with materials, but with the wealth of features that lavaland gives you, you'd be better off just leaving the shuttle in and getting rid of the asteroid field. (But if you did expand upon it as a single, on-station mining area, make sure you make a way for the crew to build into or expand into the asteroid! In the current layout it'd be blocked by either the cargo ship, the bridge, or escape!)

Aside from that, mini has a lot of good, open areas that you can dedicate towards projects, art, admin events, you name it. I'd love to see some real evolution from the old map.

optimumtact commented:
we got too many maps as is

Tlaltecuhtli commented:
> we got too many maps as is

we can always replace box 😎

trollbreeder commented:
> > we got too many maps as is
> 
> we can always replace box 😎

Dont kill Boxstation, please!

HomocidlePancakes commented:
we can always replace meta 😎

Bawhoppen commented:
am I allowed to PR updated Birdboat if we're readding Mini?

trollbreeder commented:
> am I allowed to PR updated Birdboat if we're readding Mini?

only if you update cere as well

CRITAWAKETS commented:
Remove Donut even if you dont merge this. Might aswell get rid of the inherently flawed station.

AffectedArc07 commented:
![image](https://user-images.githubusercontent.com/25063394/72389607-ad0ccd80-3720-11ea-8c0c-8a0edac308fb.png)
oops

HomocidlePancakes commented:
>Might aswell get rid of the inherently flawed station.

I agree. It’s about time we’ve deleted meta.

Syuff committed:
MiniStation map changes and tgm

Converted map file to tgm
Updated map to include experimentor and BIPIS
Removed ore smelter
Moved recycler to janitors closet
Moved portable scrubbers/pumps to hallway
Updated floor decals in departments.

cacogen commented:
Giacom's MiniStation used to be the darling of lowpop. 

I wasn't around for a lot of these newer disused maps but I wrote the following on the forums in response to a thread asking for a good map for lowpop after I'd looked back over the repo. I thought people might find it useful to know the history

>MiniStation was made by Giacom with low-pop in mind and added in 2014. It was superseded by OmegaStation in 2016, which was superseded by PubbyStation in 2018. Of the depreciated maps OmegaStation would require the least updating.  MiniStation is smallest.

cacogen commented:
@optimumtact

make a bot that stores typepaths when they change and automatically updates maps

Syuff commented:
Map updated with suggestions and converted to tgm.

> oops

Am I meant to pull from a branch of my fork instead of main? I am not familiar with systems like github.

> The asteroids to the north west of the station

I prefer the lavaland shuttle option unless me or someone else is struck with inspiration. I don't quite see the benefit of removing them so I haven't yet (lavaland shuttle is operational).

> make a bot that stores typepaths when they change and automatically updates maps

This would have saved me a lot of time in the text editor.

AffectedArc07 commented:
We already have that, it’s called CI

Speaking of which, add mini station to the test all maps list if you haven’t

AffectedArc07 reviewed (commented):
You also need to modify config/maps.txt

cacogen commented:
>We already have that

no, we don't

Syuff committed:
Update maps.txt to include ministation (votable)

AffectedArc07 commented:
We do have it, because if you change an object path without updating it, travis will fail on you. The only time maps don’t keep updated typepaths is in cases like this, where this map has been out the repo for years.

Syuff commented:
> Speaking of which, add mini station to the test all maps list if you haven’t

What/where is this?

cacogen commented:
Travis downloads and compiles the latest version of master from the command
line. If a map is included with outdated paths, it fails. The same as it
would in DreamMaker. It doesn't store old paths when they're changed and
automatically update them. It could be built into MapMerge. Wish I knew
more about coding.
-- 
Sent from my phone

Jalleo commented:
the syndicate shuttle needs removing. You also need to create a json file for ministation for it to actually get loaded up.

Syuff committed:
Updated map to include traitor objectives

removed syndicate shuttle.
created secure tech storage inside eva storage
added vault/nuke room
added all required objective_items
fixed various airlock accesses

Syuff commented:
I can't figure out what is missing that is stopping the votemap from working, the ministation.json exists in _maps folder and is now mentioned in maps.txt

Map is updated and should work properly with other game modes (traitor etc).

Ty-the-Smonk commented:
Does mini station even have a whiteship

Jalleo commented on _maps/ministation.json:
```diff
@@ -0,0 +1,11 @@
+{
+    "map_name": "MiniStation",
```
Try copying and pasting all of this into next_map.json (might be nextmap.json) While loading up let us know if it forces default to box.

Syuff replied:
Turns out map vote was working fine, just can't vote for the current map. Thanks.

Jalleo reviewed (changes requested):
This might help to get it to load up once at least.

Syuff committed:
Map bug fixes

Fixed security lockers
Relocated vending machines and protolathes for better access
Fixed bio generator

Cyberboss commented on _maps/map_files/MiniStation/job_changes.dm:
```diff
@@ -0,0 +1,150 @@
+
+/datum/job/captain/New()
+	..()
+	total_positions = 1
```
These all need `MAP_JOB_CHECK` before setting their settings or they will apply globally.

Cyberboss commented on config/maps.txt:
```diff
@@ -44,6 +44,10 @@ map kilostation
 endmap
 
+map ministation
+	votable
```
No maxplayers?

Syuff replied:
Set to 25 for now but I feel 20 would be better

Cyberboss commented on _maps/map_files/MiniStation/job_changes.dm:
```diff
@@ -0,0 +1,150 @@
+/datum/job/qm/New()
+	..()
+	total_positions = 0
+	spawn_positions = 0
```
Removed jobs should use the `MAP_REMOVE_JOB` macro. See https://github.com/tgstation/tgstation/blob/master/code/__DEFINES/maps.dm

Syuff replied:
MAP_REMOVE_JOB macro had a couple of bugs which have been addressed. The macro itself in https://github.com/tgstation/tgstation/blob/master/code/__DEFINES/maps.dm was missing a semicolon at the end of the line (unless I am using it in an unintended way?).
Removing department heads broke late round joining and I have included a mostly fix for this, detailed in https://github.com/tgstation/tgstation/pull/48788/commits/3847d2f55a91f0a60510635db88a047a1c456ccb. Although in my opinion, having late round join list be based on a dynamic list of jobs in departments instead of static would be more future proof.

Cyberboss commented:
Wasn't I supposed to make out of repo maps a thing? lmao

Syuff committed:
Map changes and code fixing

map fix traitor objectives (reactive armor)
map add teleporter to telecoms satellite
map add navigation and tracking beacons
job_changes now uses defined map macros

Syuff committed:
Missing semicolon which broke MAP_REMOVE_JOB

Syuff committed:
Fixed removing department head roles

Removing department head roles would break the late join screen.
This ensures the screen does not break, but requires adding exp_type_department to each job to fix department names not being displayed if their head is removed.

Syuff committed:
ministation maxplayers 25

optimumtact commented:
we just dont' need it.
